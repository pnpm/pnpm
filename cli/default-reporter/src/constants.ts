// In terminals, always Unix line endings are used
// even on Windows
export const EOL = '\n'
