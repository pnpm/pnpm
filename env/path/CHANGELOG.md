# @pnpm/env.path

## 1.0.0

### Major Changes

- 26b065c: Initial release.
