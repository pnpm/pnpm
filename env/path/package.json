{
  "name": "@pnpm/env.path",
  "version": "1.0.0",
  "description": "Functions for changing PATH env variable",
  "funding": "https://opencollective.com/pnpm",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib",
    "!*.map"
  ],
  "exports": {
    ".": "./lib/index.js"
  },
  "engines": {
    "node": ">=18.12"
  },
  "scripts": {
    "lint": "eslint \"src/**/*.ts\" \"test/**/*.ts\"",
    "_test": "jest",
    "test": "pnpm run compile && pnpm run _test",
    "prepublishOnly": "pnpm run compile",
    "compile": "tsc --build && pnpm run lint --fix"
  },
  "repository": "https://github.com/pnpm/pnpm/blob/main/env/path",
  "keywords": [
    "pnpm9",
    "pnpm",
    "env"
  ],
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/pnpm/pnpm/issues"
  },
  "homepage": "https://github.com/pnpm/pnpm/blob/main/env/path#readme",
  "dependencies": {
    "path-name": "catalog:"
  },
  "devDependencies": {
    "@pnpm/env.path": "workspace:*"
  }
}
