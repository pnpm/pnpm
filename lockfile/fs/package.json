{
  "name": "@pnpm/lockfile.fs",
  "version": "1.0.1",
  "description": "Read/write pnpm-lock.yaml files",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "engines": {
    "node": ">=18.12"
  },
  "files": [
    "lib",
    "!*.map"
  ],
  "scripts": {
    "lint": "eslint \"src/**/*.ts\" \"test/**/*.ts\"",
    "_test": "jest",
    "test": "pnpm run compile && pnpm run _test",
    "prepublishOnly": "pnpm run compile",
    "compile": "tsc --build && pnpm run lint --fix",
    "start": "tsc --watch"
  },
  "repository": "https://github.com/pnpm/pnpm/blob/main/lockfile/fs",
  "keywords": [
    "pnpm9",
    "pnpm",
    "shrinkwrap",
    "lockfile"
  ],
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/pnpm/pnpm/issues"
  },
  "homepage": "https://github.com/pnpm/pnpm/blob/main/lockfile/fs#readme",
  "peerDependencies": {
    "@pnpm/logger": "^5.1.0"
  },
  "devDependencies": {
    "@pnpm/lockfile.fs": "workspace:*",
    "@pnpm/logger": "workspace:*",
    "@types/js-yaml": "catalog:",
    "@types/normalize-path": "catalog:",
    "@types/ramda": "catalog:",
    "@types/semver": "catalog:",
    "@types/write-file-atomic": "catalog:",
    "tempy": "catalog:",
    "write-yaml-file": "catalog:",
    "yaml-tag": "catalog:"
  },
  "dependencies": {
    "@pnpm/constants": "workspace:*",
    "@pnpm/dependency-path": "workspace:*",
    "@pnpm/error": "workspace:*",
    "@pnpm/git-resolver": "workspace:*",
    "@pnpm/git-utils": "workspace:*",
    "@pnpm/lockfile.types": "workspace:*",
    "@pnpm/lockfile.utils": "workspace:*",
    "@pnpm/lockfile.merger": "workspace:*",
    "@pnpm/types": "workspace:*",
    "@pnpm/util.lex-comparator": "catalog:",
    "@zkochan/rimraf": "catalog:",
    "comver-to-semver": "catalog:",
    "js-yaml": "catalog:",
    "normalize-path": "catalog:",
    "ramda": "catalog:",
    "semver": "catalog:",
    "sort-keys": "catalog:",
    "strip-bom": "catalog:",
    "write-file-atomic": "catalog:"
  },
  "funding": "https://opencollective.com/pnpm",
  "exports": {
    ".": "./lib/index.js"
  }
}
