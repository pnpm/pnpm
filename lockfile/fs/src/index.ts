export {
  isEmptyLockfile,
  writeLockfiles,
  writeCurrentLockfile,
  writeWantedLockfile,
  writeLockfileFile,
} from './write'
export { existsNonEmptyWantedLockfile } from './existsWantedLockfile'
export { getLockfileImporterId } from './getLockfileImporterId'
export * from '@pnpm/lockfile.types'
export * from './read'
export { cleanGitBranchLockfiles } from './gitBranchLockfile'
export { convertToLockfileFile } from './lockfileFormatConverters'
