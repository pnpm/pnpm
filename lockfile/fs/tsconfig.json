{
  "extends": "@pnpm/tsconfig",
  "compilerOptions": {
    "outDir": "lib",
    "rootDir": "src"
  },
  "include": [
    "src/**/*.ts",
    "../../__typings__/**/*.d.ts"
  ],
  "references": [
    {
      "path": "../../packages/constants"
    },
    {
      "path": "../../packages/dependency-path"
    },
    {
      "path": "../../packages/error"
    },
    {
      "path": "../../packages/git-utils"
    },
    {
      "path": "../../packages/logger"
    },
    {
      "path": "../../packages/types"
    },
    {
      "path": "../../resolving/git-resolver"
    },
    {
      "path": "../merger"
    },
    {
      "path": "../types"
    },
    {
      "path": "../utils"
    }
  ]
}
