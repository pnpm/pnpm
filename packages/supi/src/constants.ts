export const ENGINE_NAME = `${process.platform}-${process.arch}-node-${process.version.split('.')[0]}`
export const LAYOUT_VERSION = 1
export const SHRINKWRAP_NEXT_VERSION = 4
export const SHRINKWRAP_VERSION = 3.9
