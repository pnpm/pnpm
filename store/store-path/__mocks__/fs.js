module.exports.promises = {
  mkdir: () => {},
  unlink: () => {},
  rmdir: () => {},
}
