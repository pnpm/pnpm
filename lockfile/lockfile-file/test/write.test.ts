import fs from 'fs'
import path from 'path'
import { LOCKFILE_VERSION, WANTED_LOCKFILE } from '@pnpm/constants'
import {
  readCurrentLockfile,
  readWantedLockfile,
  writeLockfiles,
} from '@pnpm/lockfile-file'
import tempy from 'tempy'
import yaml from 'yaml-tag'
import { getCurrentBranch } from '@pnpm/git-utils'

jest.mock('@pnpm/git-utils', () => ({ getCurrentBranch: jest.fn() }))

test('writeLockfiles()', async () => {
  const projectPath = tempy.directory()
  const wantedLockfile = {
    importers: {
      '.': {
        dependencies: {
          'is-negative': '1.0.0',
          'is-positive': '1.0.0',
        },
        specifiers: {
          'is-negative': '^1.0.0',
          'is-positive': '^1.0.0',
        },
      },
    },
    lockfileVersion: LOCKFILE_VERSION,
    packages: {
      '/is-negative@1.0.0': {
        os: ['darwin'],
        dependencies: {
          'is-positive': '2.0.0',
        },
        cpu: ['x86'],
        libc: ['glibc'],
        engines: {
          node: '>=10',
          npm: '\nfoo\n',
        },
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
      '/is-positive@1.0.0': {
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
      '/is-positive@2.0.0': {
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
    },
  }
  await writeLockfiles({
    currentLockfile: wantedLockfile,
    currentLockfileDir: projectPath,
    wantedLockfile,
    wantedLockfileDir: projectPath,
  })
  expect(await readCurrentLockfile(projectPath, { ignoreIncompatible: false })).toEqual(wantedLockfile)
  expect(await readWantedLockfile(projectPath, { ignoreIncompatible: false })).toEqual(wantedLockfile)

  // Verifying the formatting of the lockfile
  expect(fs.readFileSync(path.join(projectPath, WANTED_LOCKFILE), 'utf8')).toMatchSnapshot()
})

test('writeLockfiles() when no specifiers but dependencies present', async () => {
  const projectPath = tempy.directory()
  const wantedLockfile = {
    importers: {
      '.': {
        dependencies: {
          'is-positive': 'link:../is-positive',
        },
        specifiers: {},
      },
    },
    lockfileVersion: LOCKFILE_VERSION,
  }
  await writeLockfiles({
    currentLockfile: wantedLockfile,
    currentLockfileDir: projectPath,
    wantedLockfile,
    wantedLockfileDir: projectPath,
  })
  expect(await readCurrentLockfile(projectPath, { ignoreIncompatible: false })).toEqual(wantedLockfile)
  expect(await readWantedLockfile(projectPath, { ignoreIncompatible: false })).toEqual(wantedLockfile)
})

test('write does not use yaml anchors/aliases', async () => {
  const projectPath = tempy.directory()
  const wantedLockfile = {
    importers: {
      '.': {
        dependencies: {
          'is-negative': '1.0.0',
          'is-positive': '1.0.0',
        },
        specifiers: {
          'is-negative': '1.0.0',
          'is-positive': '1.0.0',
        },
      },
    },
    lockfileVersion: LOCKFILE_VERSION,
    packages: yaml`
      /react-dnd@2.5.4(react@15.6.1):
        dependencies:
          disposables: 1.0.2
          dnd-core: 2.5.4
          hoist-non-react-statics: 2.5.0
          invariant: 2.2.3
          lodash: 4.15.0
          prop-types: 15.6.1
          react: 15.6.1
        dev: false
        id: registry.npmjs.org/react-dnd/2.5.4
        peerDependencies: &ref_11
          react: '1'
        resolution:
          integrity: sha512-y9YmnusURc+3KPgvhYKvZ9oCucj51MSZWODyaeV0KFU0cquzA7dCD1g/OIYUKtNoZ+MXtacDngkdud2TklMSjw==
      /react-dnd@2.5.4(react@15.6.2):
        dependencies:
          disposables: 1.0.2
          dnd-core: 2.5.4
          hoist-non-react-statics: 2.5.0
          invariant: 2.2.3
          lodash: 4.15.0
          prop-types: 15.6.1
          react: 15.6.2
        dev: false
        id: registry.npmjs.org/react-dnd/2.5.4
        peerDependencies: *ref_11
        resolution:
          integrity: sha512-y9YmnusURc+3KPgvhYKvZ9oCucj51MSZWODyaeV0KFU0cquzA7dCD1g/OIYUKtNoZ+MXtacDngkdud2TklMSjw==
    `,
  }
  await writeLockfiles({
    currentLockfile: wantedLockfile,
    currentLockfileDir: projectPath,
    wantedLockfile,
    wantedLockfileDir: projectPath,
  })

  const lockfileContent = fs.readFileSync(path.join(projectPath, WANTED_LOCKFILE), 'utf8')
  expect(lockfileContent).not.toMatch('&')
  expect(lockfileContent).not.toMatch('*')
})

test('writeLockfiles() does not fail if the lockfile has undefined properties', async () => {
  const projectPath = tempy.directory()
  const wantedLockfile = {
    importers: {
      '.': {
        dependencies: {
          'is-negative': '1.0.0',
          'is-positive': '1.0.0',
        },
        specifiers: {
          'is-negative': '^1.0.0',
          'is-positive': '^1.0.0',
        },
      },
    },
    lockfileVersion: LOCKFILE_VERSION,
    packages: {
      '/is-negative@1.0.0': {
        // eslint-disable-next-line
        dependencies: undefined as any,
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
      '/is-positive@1.0.0': {
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
      '/is-positive@2.0.0': {
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
    },
  }
  await writeLockfiles({
    currentLockfile: wantedLockfile,
    currentLockfileDir: projectPath,
    wantedLockfile,
    wantedLockfileDir: projectPath,
  })
})

test('writeLockfiles() when useGitBranchLockfile', async () => {
  const branchName: string = 'branch'
  ;(getCurrentBranch as jest.Mock).mockReturnValue(branchName)
  const projectPath = tempy.directory()
  const wantedLockfile = {
    importers: {
      '.': {
        dependencies: {
          foo: '1.0.0',
        },
        specifiers: {
          foo: '^1.0.0',
        },
      },
    },
    lockfileVersion: LOCKFILE_VERSION,
    packages: {
      '/foo@1.0.0': {
        resolution: {
          integrity: 'sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=',
        },
      },
    },
  }

  await writeLockfiles({
    currentLockfile: wantedLockfile,
    currentLockfileDir: projectPath,
    wantedLockfile,
    wantedLockfileDir: projectPath,
    useGitBranchLockfile: true,
  })
  expect(fs.existsSync(path.join(projectPath, WANTED_LOCKFILE))).toBeFalsy()
  expect(fs.existsSync(path.join(projectPath, `pnpm-lock.${branchName}.yaml`))).toBeTruthy()
})
