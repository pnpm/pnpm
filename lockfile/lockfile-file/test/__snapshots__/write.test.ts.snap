// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`writeLockfiles() 1`] = `
"lockfileVersion: '7.0'

importers:

  .:
    dependencies:
      is-negative:
        specifier: ^1.0.0
        version: 1.0.0
      is-positive:
        specifier: ^1.0.0
        version: 1.0.0

packages:

  /is-negative@1.0.0:
    resolution: {integrity: sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=}
    engines: {node: '>=10', npm: "\\nfoo\\n"}
    cpu: [x86]
    os: [darwin]
    libc: [glibc]
    dependencies:
      is-positive: 2.0.0

  /is-positive@1.0.0:
    resolution: {integrity: sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=}

  /is-positive@2.0.0:
    resolution: {integrity: sha1-ChbBDewTLAqLCzb793Fo5VDvg/g=}
"
`;
